src/components/ViewContainer/ViewContainer.js

```diff
@@ -0,0 +1,174 @@
+  const [restoreInProgress, setRestoreInProgress] = useState(false);
+
+  const CRUDActions = getCRUDActions({
+    entityKey,
+    mutator,
+    onSuccess: networkMessage('success', entityKey, calloutRef),
+    onError: networkMessage('error', entityKey, calloutRef),
+  });
+
+  const transitionToParams = params => {
+    history.push(buildUrl(location, params));
+  };
+
+  const onShowRestoreModal = () => {
+    setShowRestoreModal(true);
+  };
+
+  const onHideRestoreModal = () => {
+    setRestoreInProgress(false);
+    setShowRestoreModal(false);
+  };
+
+  const onUpdateSuccess = (record, dispatch, properties) => {
+    const { reset: resetForm } = properties;
+
+    resetForm();
+    transitionToParams({
+      _path: `${path}/view/${record.id}`,
+      layer: null,
+    });
+  };
+
+  const deselectOnDelete = (recordId, selectItem, selectedRecords) => {
+    const isRecordSelected = selectedRecords.has(recordId);
+
+    if (isRecordSelected) {
+      selectItem(recordId);
+    }
+  };
+
+  const onDeleteSuccess = record => {
+    transitionToParams({
+      _path: `${path}/view`,
+      layer: null,
+    });
+    selectRecord(record.id);
+  };
+
+  const onDeleteError = (record, error) => {
+    if (error.status === 409) {
+      setShowExceptionModal(true);
+    }
+  };
+
+  const onRestoreSuccess = record => {
+    transitionToParams({
+      _path: `${path}/view`,
+      layer: null,
+    });
+    onHideRestoreModal();
+  };
+
+  const onRestoreError = (record, error) => {
+    onHideRestoreModal();
+  };
+
+  const restoreDefaults = async () => {
+    if (restoreInProgress) {
+      return;
+    }
+
+    const record = {};
+
+    try {
+      setRestoreInProgress(true);
+
+      await restoreDefaults.POST({});
+
+      onRestoreSuccess();
+    } catch (error) {
+      onRestoreError(record, error);
+    }
+  };
+
+  return (
+    <Fragment>
+      {children({
+        fullWidthContainer: fullWidthContainerRef.current,
+        calloutRef: calloutRef.current,
+        handleCreateSuccess: onUpdateSuccess,
+        handleEditSuccess: onUpdateSuccess,
+        handleDeleteSuccess: onDeleteSuccess,
+        handleDeleteError: onDeleteError,
+        handleRestoreSuccess: onRestoreSuccess,
+        handleRestoreError: onRestoreError,
+        ...CRUDActions,
+      })}
+      <div
+        className={sharedCss.fullWidthAndHeightContainer}
+        ref={fullWidthContainerRef}
+      />
+      <ExceptionModal
+        id={`delete-${entityKey}-exception-modal`}
+        label={<FormattedMessage id={`ui-data-import.settings.${entityKey}.exceptionModal.label`} />}
+        message={<FormattedMessage id={`ui-data-import.settings.${entityKey}.exceptionModal.message`} />}
+        showExceptionModal={showExceptionModal}
+        onClose={() => setShowExceptionModal(false)}
+      />
+      <Callout ref={calloutRef} />
+      <ConfirmationModal
+        id="restore-default-file-extensions-modal"
```

Comment:
file-extensions should not be bound to the abstract ListView.

Reply:
It will be moved back to FileExtensions